### PR TITLE
Smime_sign_regional_characters

### PR DESCRIPTION
**Pull request by ckaadic**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Currently the results of smime-sign-email is not valid for regional special characters like ä, Ø, ß and others. Fixed this by using transport encoding.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by adi88d — COMMENTED** [review]
Hi @ckaadic,
Thank you for your contribution!
can you please tell me how to reproduce the issue with `mime-sign-email` command?
i tried to run it with the special characters and it works with no erros
image



and Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for demo over DFIR.

**Comment by ckaadic** [comment]
Hello @adi88d. My initial description was missleading so I adjusted it: Currently the results of smime-sign-email is not valid for regional special characters like ä, Ø, ß and others. Fixed this by using transport encoding.

This means yes, smime-sign-email will run without showing an error. But mail gateways will not accept the smime and will not deliver the mail. This can easily be resolved by using transport encoding which I did. With a second thought it might be enough (for me) to have transport encoding as an optional argument (if someone needs it without). So I can change it to support both with an argument, if you like that more.

**Comment by ckaadic** [comment]
The send-mail command will run into a timeout with special character and will return "Mail sent successfully" when run without special character:
image

**Comment by adi88d** [comment]
Thanks @ckaadic for your response!
please add to the command the `use_transport_encoding` and change the py and yml files.

File: Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py [hunk 1]
```diff
@@ -9,8 +9,9 @@
 from tempfile import NamedTemporaryFile
 
 from charset_normalizer import from_bytes
+import quopri
 import warnings
-warnings.simplefilter("default")
+#warnings.simplefilter("default")
```
Comment (adi88d) [inline Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py:14]: revert this line

File: Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py [hunk 2]
```diff
@@ -41,17 +42,30 @@ def sign_email(client: Client, args: Dict):
     """
```
Comment (adi88d) [inline Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py:42]: please add the `use_transport_encoding: bool` arg to `sign_email` method
and update the code by checking this arg once:

> if use_transport_encoding:
     your code part
  else:
      the old code
      
in the `main` method add:
`use_transport_encoding: bool = argToBoolean(demisto.params().get('use_transport_encoding'))`
and pass it to `sign_email` method

File: Packs/SMIME_Messaging/ReleaseNotes/1_0_7.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### SMIME Messaging
+- Fixed an issue where regional characters in **message_body** lead to errors when running ***smime-sign-email***
```
Comment (adi88d) [inline Packs/SMIME_Messaging/ReleaseNotes/1_0_7.md:4]:
```suggestion
- Added the **use_transport_encoding** argument to the ***smime-sign-email*** command.
```

**Review by adi88d — CHANGES_REQUESTED** [review]
Thanks @ckaadic 
please resolve the conflicts and my comments below
and we can set up the demo at Wednesday at 12pm CET
let me know if that works for you.

File: Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py (hunk 2 above)
Comment (ckaadic) [reply Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py:42]: use_transport_encoding is not a parameter at the moment, but an argument, as you might want to change this depending on the message. I set it in the beginning of the function and use the boolean for the check afterwards.

**Comment by ckaadic** [comment]
12pm works for me. Changed two parts as asked and modified the third, as this was possible at the moment

File: Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py (hunk 2 above)
Comment (adi88d) [reply Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py:42]: @ckaadic you right its the command parameter, my bad.
what i meant in the last comment is to update the code by checking `use_transport_encoding` arg once inside `sign_email`, you check it 3 times, it should be:

> if use_transport_encoding:
> your code part
> else:
> the old code

the old part part should start from the line:
> message_body = args.get('message_body', '')
to:
> client.smime.write(out, p7, buf, SMIME.PKCS7_TEXT)

**Comment by adi88d** [comment]
please resolve the branch conflicts

**Comment by ckaadic** [comment]
changed the if part if the sign_email function as requested. Resolved conflicts and bumped the version of the Pack, as required

**Comment by ckaadic** [comment]
rebased as the "circleci: Setup Environment" step was asking for it. Still fails, but I can't find the issue. Can you give me a hint

**Review by adi88d — APPROVED** [review]
great!

**Pull request merged**